### PR TITLE
feat(frontend): support `DROP COLUMN`

### LINKED ISSUE #8351: Support column ref-count and `DROP COLUMN`.

### PR DESCRIPTION
**Pull request by BugenZhao**
Signed-off-by: Bugen Zhao <i@bugenzhao.com>I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Make the handler take an `Operation` to also support dropping columns.
- Tweak the error reporting for dropping a being-referenced column. This is naturally achieved by #8094 since we'll fail to map them with the index mapping so there's temporarily no need to implement some sort of column ref count unless we need a more detailed error message.
  - There are some false positives here since we always reference all columns when creating a materialized view now. We should fix this in #4529.
- Add some e2e tests.

Note: this is still not user-facing like #8063.
Close #8351.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/meta/src/rpc/ddl_controller.rs [hunk 1]
```diff
@@ -597,7 +596,12 @@ where
             .into_iter()
             .map(|(d, f)| Some((table_col_index_mapping.rewrite_dispatch_strategy(&d)?, f)))
             .collect::<Option<_>>()
-            .context("failed to map columns")?;
+            .ok_or_else(|| {
+                // The `rewrite` only fails if some column is dropped.
+                MetaError::invalid_parameter(
+                    "unable to drop the column due to being referenced by downstream materialized views or sinks",
```
Comment (chenzl25) [inline src/meta/src/rpc/ddl_controller.rs:602]: What about indexes?

Comment (chenzl25) [reply src/meta/src/rpc/ddl_controller.rs:602]: Theoretically, we need to drop all indexes containing the dropped column, if those indexes aren't used by other relations, but it is acceptable to throw an error for now.

Comment (BugenZhao) [reply src/meta/src/rpc/ddl_controller.rs:602]: Agree that we may need to follow the behavior of postgres in the future. I've marked this as a task in the tracking issue https://github.com/risingwavelabs/risingwave/issues/6903.

**Review by yezizp2012 — APPROVED** [review]
LGTM!

**Review by st1page — APPROVED** [review]
LGTM

**Pull request merged**